test/functional/plugin/lsp_spec.lua

```diff
@@ -2708,6 +2708,42 @@ describe('LSP', function()
       eq(expected, actual)
     end)
 
+    it('location that exceeds the line length returns end_col of line length', function()
```

Comment:
can we just modify the above test? this is a lot of (duplicate) code just to test an offset.

Reply:
happy to do that if you prefer @justinmk 

I just thought it was better to leave the test for the "happy path" intact

Reply:
that's a good practice for sure. but adding a `character` param to the `exec_lua` call in the above tests, would allow us to make one extra call with character=10000. 

so that allows testing the happy path as well as some variants. and it's clearer for others to follow, because it's only ~one call instead of 30+ lines of code :)